### PR TITLE
Feature: add hygen templates

### PR DESCRIPTION
**Pull request by robertoHeCi**
### What’s the focus of this PR
- Install Hygen and use it to generate the files needed to create a new rule.
- Update lint script

### How to review this PR
- Checkout branch
- Execute `npm i`
- Execute the command `npm run new:rule`
- Add the name of the rule
- Review the generated files

### Related work items
-

### Before submitting this PR, I made sure:
- [x] The code builds clean without any errors or warnings
- [x] I'm using our guidelines
- [] I've added tests

### CODE REVIEW

File: _templates/rule/new/test.ejs.t [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  afterEach(() => {
+    jest.resetAllMocks()
+  })
+
+  it('Example', () => {
+    global.danger = { github: gitHubMockBuilder() }
+    <%= h.changeCase.camel(name) %>()
+    expect(<%= h.changeCase.camel(name) %>).toHaveBeenCalled()
+  })
```
Comment (ddialar) [inline _templates/rule/new/test.ejs.t:27]: My proposal is to change a valid test to a failing one. For example, this one:

```
it('You must implement this testing case!!!', () => {
  global.danger = { github: gitHubMockBuilder() }
  <%= h.changeCase.camel(name) %>()
  expect(<%= h.changeCase.camel(name) %>).toHaveBeenCalled()
  expect(false).toBeThruty()
})
```

File: .eslintignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 dist/
-jest.config.ts
\ No newline at end of file
+jest.config.ts
+_templates
```
Comment (ddialar) [inline .eslintignore:3]: If you remove the templates from the repo, what is going to be used by Hygen to show the different options in the CLI?

Comment (robertoHeCi) [reply .eslintignore:3]: This file is the `.eslintignore`, not the `.gitignore` 😄

File: _templates/rule/new/test.ejs.t (hunk 1 above)
Comment (robertoHeCi) [reply _templates/rule/new/test.ejs.t:27]: I like it!!

File: .eslintignore (hunk 2 above)
Comment (ddialar) [reply .eslintignore:3]: Ok... I'm going to cry under my bed 😢

**Review by ddialar — APPROVED** [review]

**Pull request merged**